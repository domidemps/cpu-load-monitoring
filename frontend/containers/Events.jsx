/** @jsx jsx */
import {css, jsx} from '@emotion/core'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'
import {COLORS} from '../styles/material_ui_raw_theme_file'
import {useSelector} from 'react-redux'
import {darken} from '@material-ui/core'

import isEmpty from 'lodash/isEmpty'
import countBy from 'lodash/countBy'
import get from 'lodash/get'

const styles = css`
  width: 60%;
  margin: 25px;
  padding: 15px;
  .alignLeft {
    text-align: left;
  }
  .alignCenter {
    text-align: center;
  }
  .heavyLoad {
    color: ${COLORS.heavyLoad};
  }
  .normalLoad {
    color: ${COLORS.normalLoad};
  }
  .defaultText {
    margin: 20px;
    color: ${darken(COLORS.text, 0.4)};
  }
`

export default function Events() {
  const events = useSelector(state => state.load.events)
  const occurrences = countBy(events, event => {
    return event.type
  })
  return (
    <Paper elevation={5} css={styles}>
      <div className="alignCenter">
        <Typography variant="h5">Events - Heavy CPU loads</Typography>
        {isEmpty(events) ? (
          <Typography variant="h6" className="defaultText">
            <i>No events to display</i>
          </Typography>
        ) : (
          <Typography variant="h6" className="alignLeft">
            <span className="heavyLoad">{get(occurrences, 'heavy', 0)} total</span>
            <br />
            <span className="normalLoad">{get(occurrences, 'recovery', 0)} recovered</span>
          </Typography>
        )}
      </div>
    </Paper>
  )
}
