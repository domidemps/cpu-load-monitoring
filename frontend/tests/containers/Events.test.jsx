import React from 'react'
import renderer from 'react-test-renderer'
import Events from '../containers/Events'


test('Events renders correctly', () => {
  const component = renderer.create(
    <Events/>
  )
})