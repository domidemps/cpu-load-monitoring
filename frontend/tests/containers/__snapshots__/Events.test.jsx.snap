// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events renders correctly if no events 1`] = `
<div
  className="MuiPaper-root css-h168i7-Events MuiPaper-elevation5 MuiPaper-rounded"
>
  <div
    className="alignCenter"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Events
    </h5>
    <h6
      className="MuiTypography-root defaultText MuiTypography-h6"
    >
      <i>
        No events to display
      </i>
    </h6>
    <div
      className="scrollable"
    />
  </div>
</div>
`;

exports[`Events renders correctly the list of events 1`] = `
<div
  className="MuiPaper-root css-h168i7-Events MuiPaper-elevation5 MuiPaper-rounded"
>
  <div
    className="alignCenter"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5"
    >
      Events
    </h5>
    <div
      className="bigSpaceAround"
    >
      <span
        className="heavyLoad"
      >
        1
         heavy CPU loads, 
      </span>
      <span
        className="normalLoad"
      >
        1
         recovered
      </span>
    </div>
    <div
      className="scrollable"
    >
      <div
        className="MuiPaper-root MuiAlert-root MuiAlert-standardError spaceAround MuiPaper-elevation0"
        role="alert"
      >
        <div
          className="MuiAlert-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          className="MuiAlert-message"
        >
          System under heavy CPU load from 12:30:00 to 12:33:00
        </div>
      </div>
      <div
        className="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess spaceAround MuiPaper-elevation0"
        role="alert"
      >
        <div
          className="MuiAlert-icon"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          className="MuiAlert-message"
        >
          System recovered from heavy CPU load since 12:33:10
        </div>
      </div>
    </div>
  </div>
</div>
`;
